src/Cli/func/Helpers/DotnetHelpers.cs

```diff
@@ -283,87 +281,60 @@ private static async Task TemplateOperationAsync(Func<Task> action, WorkerRuntim
 
             if (workerRuntime == WorkerRuntime.DotnetIsolated)
             {
-                await EnsureIsolatedTemplatesInstalled();
+                await EnsureIsolatedTemplatesInstalled(action);
             }
             else
             {
-                await EnsureWebJobsTemplatesInstalled();
+                await EnsureWebJobsTemplatesInstalled(action);
             }
-
-            await action();
         }
 
-        private static async Task EnsureIsolatedTemplatesInstalled()
+        private static async Task EnsureIsolatedTemplatesInstalled(Func<Task> action)
         {
-            if (AreDotnetTemplatePackagesInstalled(await _installedTemplatesList.Value, WebJobsTemplateBasePackId))
+            try
             {
+                // Uninstall any existing webjobs templates, as they conflict with isolated templates
                 await UninstallWebJobsTemplates();
```

Comment:
The template package name here is a bit misleading. These are actually in-proc templates. Would be good if we can rename this to inproc templates.